runtime/mux.go

```diff
@@ -270,13 +290,13 @@ func NewServeMux(opts ...ServeMuxOption) *ServeMux {
 		opt(serveMux)
 	}
 
-	if serveMux.incomingHeaderMatcher == nil {
-		serveMux.incomingHeaderMatcher = DefaultHeaderMatcher
-	}
-
-	if serveMux.outgoingHeaderMatcher == nil {
-		serveMux.outgoingHeaderMatcher = func(key string) (string, bool) {
-			return fmt.Sprintf("%s%s", MetadataHeaderPrefix, key), true
+	for matcher, fallback := range map[*HeaderMatcherFunc]HeaderMatcherFunc{
+		&serveMux.incomingHeaderMatcher:  DefaultHeaderMatcher,
+		&serveMux.outgoingHeaderMatcher:  defaultOutgoingHeaderMatcher,
+		&serveMux.outgoingTrailerMatcher: defaultOutgoingTrailerMatcher,
+	} {
+		if *matcher == nil {
+			*matcher = fallback
```

Comment:
This is clever, but I think I'd prefer that it was just 3 if statements along the line of the code you replaced 🙏🏻.